File: UI/Converters/SingleLineTextConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;

namespace UI.Converters {

    [ValueConversion(typeof(string), typeof(string))]
    public class SingleLineTextConverter : IValueConverter {

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if (value == null)
                return value;

            var input = (string)value;
            input = input.Replace("\n", " ").Replace("\r", "");
            if (input.Length > 60)
                input = input.Substring(0, 60) + "...";
            return input;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotSupportedException("Cannot convert a single line of text back to multiline input.");
        }
    }
}

File: UI/Converters/SingleLineTextConverter.fs
namespace UI.Converters

open System
open System.Collections.Generic
open System.Globalization
open System.Windows
open System.Windows.Controls
open System.Windows.Data
open System.Windows.Media

[<ValueConversion(typeof<string>, typeof<string>)>]
type SingleLineTextConverter() =

    interface IValueConverter with

        member x.Convert(value:obj, targetType:Type, parameter:obj, culture:CultureInfo) =
            match value with
            | :? string as text ->
                let text = text.Replace("\n"," ")
                let text = text.Replace("\r","")
                if text.Length > 60 then text.Substring(0,60) + "..."
                else text
                :> obj
            | x -> x
        
        member x.ConvertBack(value:obj, targetType:Type, parameter:obj, culture:CultureInfo) =
            raise (new NotImplementedException("Cannot convert a single line of text back to multiline input."))